nvm.sh

```diff
@@ -695,6 +695,82 @@ nvm_binary_available() {
   nvm_version_greater_than_or_equal_to "$(nvm_strip_iojs_prefix "${1-}")" v0.8.6
 }
 
+nvm_set_colors() {
+  if [ nvm_has_colors ]
+  then
+    COLORS=$1
+    INSTALLED_COLOR="$(nvm_print_color_code ${COLORS:0:1})"
+    LTS_AND_SYSTEM_COLOR="$(nvm_print_color_code ${COLORS:1:1})"
+    CURRENT_COLOR="$(nvm_print_color_code ${COLORS:2:1})"
+    NOT_INSTALLED_COLOR="$(nvm_print_color_code ${COLORS:3:1})"
+    DEFAULT_COLOR="$(nvm_print_color_code ${COLORS:4:1})"
+    printf "Setting colors to: \033[${INSTALLED_COLOR}${COLORS:0:1}\033[${LTS_AND_SYSTEM_COLOR}${COLORS:1:1}\033[${CURRENT_COLOR}${COLORS:2:1}\033[${NOT_INSTALLED_COLOR}${COLORS:3:1}\033[${DEFAULT_COLOR}${COLORS:4:1}\033[0m"
```

Comment:
```suggestion
    nvm_echo "Setting colors to: \033[${INSTALLED_COLOR}${COLORS:0:1}\033[${LTS_AND_SYSTEM_COLOR}${COLORS:1:1}\033[${CURRENT_COLOR}${COLORS:2:1}\033[${NOT_INSTALLED_COLOR}${COLORS:3:1}\033[${DEFAULT_COLOR}${COLORS:4:1}\033[0m"
```

Reply:
@ljharb, `nvm_echo` does not expand nested function calls in the variables we're passing to it. We tried setting a variable to concatenate the output of a function with the expanded value of other variables and then passing that into nvm_echo with the same result, so went back to using `printf`. Do you have any suggestions on how to get `nvm_echo` to work here?

Reply:
@ljharb We made a new function that mirrors `nvm_echo` but uses `%b` instead of `%s`.